defold-rive/src/comp_rive.cpp

```diff
@@ -226,11 +217,161 @@ namespace dmRive
+            return 0;
+        }
+
+        dmGameObject::SetBone(instance, true);
+        return instance;
+    }
+
+    static bool CreateGameobjects(RiveWorld* world, RiveComponent* component)
+    {
+        dmGameObject::HInstance rive_instance            = component->m_Instance;
+        dmGameObject::HCollection collection             = dmGameObject::GetCollection(rive_instance);
+        dmRive::RiveSceneData* data                      = (dmRive::RiveSceneData*) component->m_Resource->m_Scene->m_Scene;
+        rive::File* f                                    = data->m_File;
+        rive::Artboard* artboard                         = f->artboard();
+        const std::vector<rive::Core*>& artboard_objects = artboard->objects();
+
+        artboard->advance(0.0f);
+
+        // transform index to object index
+        dmArray<uint32_t> transform_to_object_array;
+        transform_to_object_array.SetCapacity(64);
+
+        // object index to transform index
+        dmHashTable<uint32_t, uint32_t> transform_to_object_map;
+        transform_to_object_map.SetCapacity(32, 128); // Todo: What are good values here?
+
+        for (int i = 0; i < artboard_objects.size(); ++i)
+        {
+            rive::Core* object = artboard_objects[i];
+            if (object->is<rive::TransformComponent>())
+            {
+                if (transform_to_object_map.Full())
+                {
+                    transform_to_object_map.SetCapacity(32, transform_to_object_map.Capacity() + 128);
+                }
+
+                if (transform_to_object_array.Full())
+                {
+                    transform_to_object_array.OffsetCapacity(64);
+                }
+
+                transform_to_object_map.Put(i, transform_to_object_array.Size());
+                transform_to_object_array.Push(i);
+            }
+        }
+
+        int node_count = transform_to_object_array.Size();
+        component->m_NodeInstances.SetCapacity(node_count);
+        component->m_NodeInstances.SetSize(node_count);
+        component->m_NodeInstanceIds.SetCapacity(node_count);
+        component->m_NodeInstanceIds.SetSize(node_count);
+        component->m_NodeInstanceToObjectIndex.SetCapacity(node_count);
+        component->m_NodeInstanceToObjectIndex.SetSize(node_count);
+
+        if (node_count > world->m_ScratchInstances.Capacity())
+        {
+            world->m_ScratchInstances.SetCapacity(node_count);
+        }
+
+        dmGameObject::HInstance root_instance = MakeBoneInstance(collection);
+        if (root_instance == 0x0)
+        {
+            return false;
+        }
+
+        dmVMath::Vector3 inv_scale(1.0f, -1.0f, 1.0f);
+        dmVMath::Vector3 origin_position(-artboard->width()/2, artboard->height()/2, 0);
+
+        dmGameObject::SetScale(root_instance, inv_scale);
+        dmGameObject::SetPosition(root_instance, Point3(origin_position));
+        dmGameObject::SetParent(root_instance, rive_instance);
```

Comment:
This is for conforming the coordinate system with Defold. Rive's coordinates have 0,0 at the top left corner of an artboard, but we want it to be centered around an image.